include/aws/common/string.h

```diff
@@ -205,6 +205,12 @@ struct aws_byte_cursor aws_byte_cursor_from_string(const struct aws_string *src)
 AWS_COMMON_API
 struct aws_string *aws_string_clone_or_reuse(struct aws_allocator *allocator, const struct aws_string *str);
 
+/* Computes the length of a c string in bytes assuming the character set is either ASCII or UTF-8. If no NULL character
+ * is found within max_read_len of str, AWS_ERROR_C_STRING_BUFFER_NOT_NULL_TERMINATED is raised. Otherwise, str_len
+ * will contain the string length minus the NULL character, and AWS_OP_SUCCESS will be returned. */
+AWS_COMMON_API
+int aws_c_string_secure_strlen(const char *str, size_t max_read_len, size_t *str_len);
```

Comment:
Just to bikeshed on naming a bit, I think having "c_string" and "strlen" in the name is redundant. Maybe just `aws_secure_strlen` (matches `aws_secure_zero`) or `aws_c_string_secure_length` (matches stuff in this file) instead?

Reply:
sure

Reply:
+1 aws_secure_strlen()
or just aws_strlen()